src/providers/data-store/data-store.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { SecureStorageObject } from '@ionic-native/secure-storage';
+import { NetworkStateProvider } from '../network-state/network-state';
+
+@Injectable()
+export class DataStoreProvider {
+
+  defaultStoreName: string = 'MES';
+
+  secureContainer: SecureStorageObject = null;
+
+  constructor(public networkState: NetworkStateProvider) {
+  }
+
+  setSecureContainer(container: SecureStorageObject): void {
+    this.secureContainer = container;
+  }
+
+  getSecureContainer(): SecureStorageObject {
+    return this.secureContainer;
+  }
+
+  getKeys(): any {
+    if (!this.secureContainer) {
+      return Promise.resolve('getKeys');
```

Comment:
Blank array would probably be better here and then you can put a type other than `any` on the method (keeps the type uniform).

Reply:
done